worker/detectors/features.go

```diff
@@ -54,15 +56,24 @@ func RegisterFeaturesDetector(name string, f FeaturesDetector) {
 }
 
 // DetectFeatures detects a list of FeatureVersion using every registered FeaturesDetector.
-func DetectFeatures(data map[string][]byte) ([]database.FeatureVersion, error) {
+func DetectFeatures(data map[string][]byte, namespaces []database.Namespace) ([]database.FeatureVersion, error) {
 	var packages []database.FeatureVersion
 
 	for _, detector := range featuresDetectors {
-		pkgs, err := detector.Detect(data)
-		if err != nil {
-			return []database.FeatureVersion{}, err
+		for _, namespace := range namespaces {
+			if detector.Supported(namespace) {
+				pkgs, err := detector.Detect(data)
+				if err != nil {
+					return []database.FeatureVersion{}, err
+				}
+				// Ensure that every feature has a Namespace associated
```

Comment:
Do you mean the 'Namespace' value? It is not set in featureDetectors. 
In the previous implementation, it was filled by work.go/detectFeatures.
Since we support multiple namespaces, I move that codes here.